frontend/src/metabase/admin/settings/components/UploadSettings/UploadSettings.unit.spec.tsx

```diff
@@ -566,5 +572,27 @@ describe("Admin > Settings > UploadSetting", () => {
     expect(
       screen.getByText(/uploads to the Sample Database are for testing only/i),
     ).toBeInTheDocument();
+
+    expect(
+      screen.queryByText("Additional terms apply."),
+    ).not.toBeInTheDocument();
+  });
+
+  it("should show an extended warning for h2 databases on hosted instances", async () => {
+    setup({ isHosted: true });
+    await userEvent.click(await screen.findByText("Select a database"));
+
+    await userEvent.click(await screen.findByText("Db Cinco")); // h2
+
+    expect(
+      screen.getByText(/uploads to the Sample Database are for testing only/i),
+    ).toBeInTheDocument();
+
+    await userEvent.hover(screen.getByText("Additional terms apply."));
+    expect(
+      within(screen.getByRole("tooltip")).getByText(
+        /By enabling uploads to the Sample Database, you agree that you will not upload or otherwise transmit any individually identifiable information/,
+      ),
+    ).toBeInTheDocument();
```

Comment:
I _think_ if you put this test before the previous one, that one will fail. I don't think we should be changing token features within the same test file. It's a trap for the next guy who adds a test below this one.  we should create separate enterprise/oss files for this.